### ISSUE TITLE
Feature request: option to combine saving into file and automatically copy into buffer

### ISSUE THREAD

**Issue by mikkovedru** [flameshot-org/flameshot#2750]
### Feature Description

There are already options to:
- "Copy URL after upload"
- "Copy file path after save"

I want a feature combining saving the file and copying the image into a buffer. The current "Save image after copy" is not providing what I want. What the current feature does:
- Good: copying of the image in the buffer
- Bad: saving the file. It is automatic and asks nothing (not the filename nor the format). It just saves in the default directory under the default name. This is not good enough as my "default name format" only prepends the ISO date while the actual name must be written by me. The default directory is also only a starting point; the actual directory the image is saved in is chosen manually.

What I would like to have:
1. I choose the save image option and receive the usual save box in which I can:
	1. change the directory
	2. change the format (.jpg, instead of .png)
	3. change the name
2. After confirmation the image is saved as well as copied in the buffer in the best lossless .png format (regardless of what format I chose to save my file in)

Thank you for your consideration. Big thanks for such a marvelous program!

**Comment by mmahmoudian** [comment]
Have you read our CLI page, especially the Final actions?

https://flameshot.org/docs/advanced/commandline-options/#final-actions

You can combine all final actions together as long as they don't conflict with each other. For instance the `--raw` and `--print-geometry` both use stdout which conflict

**Comment by mikkovedru** [comment]
I did not know about that new feature. Thank you for linking.

Unfortunately, it doesn't solve the issue. What I want:
1. One hotkey to run `flameshot gui` command with the ability to do all the annotating I want.
2. After that I want to be able to choose one of:
	2.1. Upload
	2.2. Pin
	2.3. Copy to clipboard (without saving the file)
	2.4. Save to file (with the ability to choose both the format and the directory) **which also automatically copies image to clipboard**

This is not possible to do because
1. I want to choose my action and not simply accept it (can't have a bunch of different hotkeys for every single option and combination of them)
![image](https://user-images.githubusercontent.com/2521942/178104468-814cb039-282e-4955-81f6-7b1e3e71449f.png)
3. I want to choose my action after annotating, not before

### Additional
Now that I think of it, combining upload with locally saving the file (or the opposite) would also be convenient.

Perhaps this is an opportunity to rethink how to do this aspect of the program better, especially when it comes to the graphical interface. Creating a bunch of checkboxes that are contradicting each other and adding them into the already big list of various checkboxes is a mess. There must be a better way to structure it. I think that I have a solution and it involved a combination of:
1. A separate "linked actions" group in settings (to decrease the already long list in General settings)
2. Several layers/subcategories/subsettings
3. Radiobuttons (if settings contradict each other).


Here is what I propose most of the settings would look like (including the already existing options):

---
- Pin
	- "Save image after pin"
		- "Save image automatically" (disabled if the parent is not selected)
	- "After action" (**radiobuttons**)
		- "Copy to clipboard after pin"
		- "Copy file path after upload and save" (disabled if "Save image after pin" is not chosen)	
- Upload
	- "Upload image without confirmation"
	- "Save image after upload"
		- "Save image automatically" (disabled if the parent is not selected)
	- "After action" (**radiobuttons**)
		- "Copy URL after upload"
		- "Copy to clipboard after upload"
		- "Copy file path after upload and save" (disabled if "Save image after upload" is not chosen)
- Copy to clipboard
	- "Save image after copy"
		- "Save image automatically" (disabled if the parent is not selected)
- Save to file
	- "Save image automatically"
	- "After action" (**radiobuttons**)
		- "Copy file path after save"
		- "Copy to clipboard after save"
---

What do you think?

**Comment by mmahmoudian** [comment]
What you proposed is a multi-step multi-menu hierarchy which frankly would brutally slow-down a quick screenshotting process. Perhaps this can be added only to the Launcher window under an Advanced button or something.

I can kinda see your point, but until we have a solid approach to address this from design perspective, I wouldn't step a foot in this direction. Such change can break lots of people's scripts and workflow, plus the usecase you mentioned is a bit uncommon. We have had only one other user complaining about this.

May I suggest handling this situation [for now] with a small and short script? You can have a few line long script to take a screenshot and store it in a temp file, then bring a menu system like `rofi`, or `fzf` with multiselect enabled and easily pull this off. The only part that cannot be pulled off through a script [yet] is pinning which any of the `pqiv`, `imv`, `feh`, or `display` can substitute for that.

**Comment by mikkovedru** [comment]
I don't see how my suggestion would slow down a quick screenshotting process. It doesn't influence the screenshotting process in any other way than :
1. making it easier to understand what is happening (in Settings)
2. (voluntary) streamlines a more complex workflow by allowing to batch several functions into one.

Nothing changes for the worse if one only uses the existing features.

My suggestion is two-fold:
1. (my initial comment) add one/two of the features similar to the ones flameshot already has ("Copy URL after upload", "Copy file path after save", "Save image after copy")
2. (the last part of the last comment) improve the Settings by **decreasing** the complexity, by improving understandability.

Currently, the General settings list is already so long, that it makes it **a bit** too confusing, especially because they are thrown together in random order no matter how closely or loosely they relate to each other.

My second suggestion was to organize the Settings list better, to group similar items together and make everything more consistent.



The lack of this feature is not a deal-breaker for me, even though I am still adamant that my suggestion - even if only when it comes to changes to Settings - would be an improvement. Would be nice if I was able to explain my opinion better. But it's ok if you still disagree. In this case, I suggest we leave this suggestion open and:
1. wait for other people's comments for new perspectives
2. wait until the program and the settings menu becomes even longer and more clearly (in contrast to the current "**a bit** too confusing") complicated. Perhaps at that point, you will remember my suggestion, see it with new eyes and will find it useful in some capacity.

Thank you for your answer and a practical solution.

**Comment by mikkovedru** [comment]
I made a PR improving the randomness of settings by reordering items and improving descriptions: #2777

**Comment by mmahmoudian** [comment]
> Currently, the General settings list is already so long, that it makes it a bit too confusing, especially because they are thrown together in random order no matter how closely or loosely they relate to each other.

I agree. It has organically developed and increased and needs an overhaul.

> I don't see how my suggestion would slow down a quick screenshotting process

Sorry, I have misunderstood your suggestion. i thought you have proposed to have a window open at the end point of the `flameshot gui` process that user can select how to chain actions together.

On second though and after a bad sleep, what do you think of having a button which user can customize and bundle multiple final actions together? Does it address the first part of your feature request?

**Comment by borgmanJeremy** [comment]
@mmahmoudian You are proposing a custom button that users can config to any number of final actions? That could work.

**Comment by mmahmoudian** [comment]
> @mmahmoudian You are proposing a custom button that users can config to any number of final actions? That could work. 

Yes that's what I was thinking. This way user can hats a personalized combo action to perform routine combo tasks without using CLI. This button should be hidden when CLI final actions are defined. I personally wouldn't probably use it, but I can see how it can be a QoL feature esp. Cot for less technical people.

**Comment by mikkovedru** [comment]
> @mmahmoudian On second though and after a bad sleep, what do you think of having a button which user can customize and bundle multiple final actions together? Does it address the first part of your feature request?

This feature seems useful, especially if there will be an opportunity to have multiple "Bundled Action Number 1" buttons with hotkeys assigned to them.

But from purely my point of view, this is a clear overkill for what I was requesting. I was asking for something really simple, something akin to what is already implemented in the program. The program already has:
1. "Copy URL after upload"
2. "Copy file path after save"
and what I was asking was to implement:
3. "Copy file after save".
